torch/_dynamo/decorators.py

```diff
@@ -74,8 +74,40 @@ def assume_constant_result(fn):
 
 def allow_in_graph(fn):
     """
-    Customize which functions TorchDynamo will include in the generated
-    graph. Similar to `torch.fx.wrap()`.
+    Tells the compiler frontend (Dynamo) to skip symbolic introspection of the function
+    and instead directly write it to the graph when encountered.
+
+    If you are trying to black-box a Python function for use with torch.compile
+    (aka something similar to torch.fx.wrap),
+    do not use this API. Instead, please create a custom operator:
+    https://docs.google.com/document/d/1_W62p8WJOQQUzPsJYa7s701JXt0qf2OfLub2sbkHOaU/edit
+
+    The main use case for :func:`allow_in_graph()` is as an escape hatch for the compiler frontend:
+    if you know the function works w.r.t. to the downstream components of the compilation
+    stack (AOTAutograd and Inductor) but there is a Dynamo bug that prevents it from
+    symbolically introspecting the function properly, then one can decorate said function
+    with allow_in_graph to bypass Dynamo.
```

Comment:
I don't think the usecase here is only to work around bugs.  I'd suggest adding something like:

| Mechanism      | Frontend (dynamo/symbolic_trace) | Backend (AOTAutograd) |
| -------------- | -------------------- | -------------------- |
| no decorator   | trace inside         | trace inside         | 
| allow_in_graph | opaque callable      | trace inside         |
| wrap (for FX) | opaque callable      | trace inside         |
| custom ops     | opaque callable      | opaque callable      |

Which makes the differences more clear.

Reply:
Sounds good, I'll add a table

Reply:
Jason: sometimes people want to execute stuff at compile time: allow_in_graph as a constant. e.g. we allow_in_graph einops (implementation is traceable: string parsing)

Jason is neutral on the rename.